FogCreek/Glitch-Community
# Judicious pruner: componentize project items in search results

## Links
* Remix link: https://judicious-pruner.glitch.me
* Manuscript link: https://glitch.manuscript.com/f/cases/3328196/

## GIF/Screenshots

### Search results (logged in)
![Screen Shot 2019-04-02 at 2 29 36 PM](https://user-images.githubusercontent.com/938722/55427730-a2989300-5555-11e9-9e46-af6e6babf62f.png)

### Project with many users
![Screen Shot 2019-04-02 at 2 29 12 PM](https://user-images.githubusercontent.com/938722/55427771-b80dbd00-5555-11e9-8b74-cbb9f4c79b7c.png)

### Private project
(Note that private projects don't actually come up in search, but this component is intended to eventually work in other contexts)
![Screen Shot 2019-04-02 at 2 38 52 PM](https://user-images.githubusercontent.com/938722/55427787-bfcd6180-5555-11e9-8f56-a5b90a25eee6.png)

## Changes
- make new ProjectItem component (like ProjectItem presenter, but without the overrides for collection views)
- use new component in search results
- added gentle fade on overflowing UsersList

## How To Test:
Explore search results, e.g. https://judicious-pruner.glitch.me/search?q=community-staging, and verify that the results look "normal".

## Feedback I'm looking for:
- does this look & work correctly?
- does this have unforseen side effects on other project views?

## Next steps (for future PRs)
-  use in other project item contexts (integrate into or after user page search PR)
-  create project item component that includes notes, exclude descriptions, for use in collection pages
-  componentize its non-component dependencies (e.g. ProjectOptionsPop, UsersList, etc) and remove the `:global` css overrides
-  delete the old project item presenter
